pkg/lib/operation/project/local/template/delete/operation.go

```diff
@@ -20,6 +24,47 @@ type dependencies interface {
 	StorageApi() (*storageapi.Api, error)
 }
 
-func Run(projectState *project.State, branch model.BranchKey, instance string, d dependencies) error {
+func Run(projectState *project.State, branch model.BranchKey, instance string, o Options, d dependencies) error {
+	logger := d.Logger()
+
+	// Get plan
+	plan, err := deleteTemplate.NewPlan(projectState.State(), branch, instance)
+	if err != nil {
+		return err
+	}
+
+	// Log plan
+	if !plan.Empty() {
+		plan.Log(logger)
+	}
+
+	if !plan.Empty() {
+		// Dry run?
+		if o.DryRun {
+			logger.Info("Dry run, nothing changed.")
+			return nil
+		}
+
+		// Invoke
+		if err := plan.Invoke(projectState.Ctx(), projectState.LocalManager()); err != nil {
+			return utils.PrefixError(`cannot delete template configs`, err)
+		}
+
+		// Remove template instance from metadata
+		branchState := projectState.GetOrNil(branch).(*model.BranchState)
+		if err := branchState.Local.Metadata.DeleteTemplateUsage(instance); err != nil {
+			return utils.PrefixError(`cannot remove template instance metadata`, err)
+		}
+		saveOp := projectState.LocalManager().NewUnitOfWork(projectState.Ctx())
+		saveOp.SaveObject(branchState, branchState.LocalState(), model.NewChangedFields())
+
+		// Save manifest
+		if _, err := saveManifest.Run(projectState.ProjectManifest(), projectState.Fs(), d); err != nil {
+			return err
+		}
```

Comment:
@webrouse Tak jsem opět marnej. 😕 Okopíroval jsem tady ten update branch metadat z `template-use` operace kde jsi to taky zachraňoval (https://github.com/keboola/keboola-as-code/blob/57493b17ccc36f9c1fef8777a5013ab1889d92ba/pkg/lib/operation/project/local/template/use/operation.go#L88-L101) a stejně mi to nefunguje. V E2E testu se má smazat jedna instance ze dvou v metadatech ale nestane se to. Mám na `branchState.Local.Metadata.DeleteTemplateUsage()` funkční unit test i jsem ověřil že se z `branchState.Local.Metadata` opravdu smažou, ale třeba v `projectState.MainBranch().Metadata` zůstanou. (Mně teda upřímně řečeno pořád ještě uniká jak/kde se to má sesynchronizovat tak, aby se to propsalo až na ten disk. 🙈)
![CleanShot 2022-05-04 at 17 47 36](https://user-images.githubusercontent.com/215660/166720448-9619b6ee-b69d-4e84-8fcb-18fdc0082bd6.jpg)

Reply:
idem to pozriet

Reply:
@JakubMatejka myslim, ze ti tam chyba `saveOp.Invoke()`, neskusal som to, ak to nepomoze,pozrietm to podrobnejsie.

BTW to mazanie metadat z branch by asi tiez mohlo byt v `plan.Invoke`.

Reply:
jo, to chybělo. Díky 👍 Už je to funkční.